addons/website/static/src/snippets/s_table_of_content/options.js

```diff
@@ -111,6 +111,12 @@ options.registry.TableOfContent = options.Class.extend({
         const exception = (tocEl) => !tocEl.querySelector('.s_table_of_content_navbar a');
         this._activateScrollSpy(exception);
         this.oldHeadingsEls = [...headingsEls.map(el => el.cloneNode(true))];
+        const blockTextContent = this.$target[0].textContent.replace(/\s*\n\s*/g, '')
+        if (targetedEls.length === 0 && blockTextContent === "") {
+            // destroy public widget and remove the ToC since there are no more
+            // child elements.
+            this.trigger_up('remove_snippet', {$snippet: this.$target});
+        }
```

Comment:
I have tried this way, and we still need to check `targetedEls.length` else as soon as we remove the Heading snippet gets deleted.